packages/runtime/test/consumption/attributes-e2e.test.ts

```diff
@@ -1001,3 +1004,53 @@ describe("Get (shared) versions of attribute", () => {
         });
     });
 });
+
+describe(DeletePeerSharedAttributeUseCase.name, () => {
+    test("should delete a peer shared identity attribute", async () => {
+        const sOSIA = await executeFullCreateAndShareRepositoryAttributeFlow(services1, services2, {
+            content: {
+                value: {
+                    "@type": "GivenName",
+                    value: "Petra Pan"
+                },
+                tags: ["tag1", "tag2"]
+            }
+        });
+
+        const rPSIA = (await services2.consumption.attributes.getAttribute({ id: sOSIA.id })).value;
+        expect(rPSIA).toBeDefined();
+
+        const deletionResult = await services2.consumption.attributes.deletePeerSharedAttribute({ attributeId: sOSIA.id });
+        expect(deletionResult.isSuccess).toBe(true);
+
+        const getDeletedAttributeResult = await services2.consumption.attributes.getAttribute({ id: sOSIA.id });
+        expect(getDeletedAttributeResult).toBeAnError(/.*/, "error.runtime.recordNotFound");
+    });
+
+    test("should notify about identity attribute deletion by peer", async () => {
+        const sOSIA = await executeFullCreateAndShareRepositoryAttributeFlow(services1, services2, {
+            content: {
+                value: {
+                    "@type": "GivenName",
+                    value: "Petra Pan"
+                },
+                tags: ["tag1", "tag2"]
+            }
+        });
+
+        const deletionTime = CoreDate.utc();
+        const notification = (await services2.consumption.attributes.deletePeerSharedAttribute({ attributeId: sOSIA.id })).value;
+        await syncUntilHasMessageWithNotification(services1.transport, notification.id);
+        await services1.eventBus.waitForEvent(AttributeDeletedByPeerEvent, (e) => {
+            return e.data.id.toString() === sOSIA.id;
+        });
+
+        const result = await services1.consumption.attributes.getAttribute({ id: sOSIA.id });
+        expect(result.isSuccess).toBe(true);
+        const updatedAttribute = result.value;
+        expect(updatedAttribute.deletionStatus?.deletedByPeer).toBeDefined();
+        expect(CoreDate.from(updatedAttribute.deletionStatus!.deletedByPeer!).isSame(deletionTime, "second")).toBe(true);
+    });
```

Comment:
The functionality mentioned in the TODO is not implemented yet. Probably, this will be implemented in another PR. I'll keep the TODO comments as a reminder, though, since the branch will not be merged in main, but in a dedicated feature branch. Only merging that feature branch in main, all TODO comments should be dismissed.